File: entrada/archivo4.c
#include <stdio.h>



#define MAX_STUDENTS 3
#define MAX_SUBJECTS 3

struct Student {
    char name[50];
    int id;
    float grades[MAX_SUBJECTS];
    float average;
};

void calculateAverage(struct Student *student) {
    float sum = 0;
    for (int i = 0; i < MAX_SUBJECTS; i++) {
        sum += student1->grades[i];
    }
    student->average = sum / MAX_SUBJECTS;
}

int main() {
    struct Student students[MAX_STUDENTS];

    for (int i = 0; i < MAX_STUDENTS; i++) {
        printf("Enter name of student %d: ", i + 1);
        scanf("%s", students[i].name);

        printf("Enter id of student %d: ", i + 1);
        scanf("%d", &students[i].id);

        printf("Enter grades for student %d:\n", i + 1);
        for (int j = 0; j < MAX_SUBJECTS; j++) {
            printf("Enter grade for subject %d: ", j + 1);
            scanf("%f", &students[i].grades[j]);
        }

        calculateAverage(&students[i]);
    }

    printf("\n\nDisplaying Information:\n\n");
    for (int i = 0; i < MAX_STUDENTS; i++) {
        printf("Name: %s\n", students[i].name);
        printf("ID: %d\n", students[i].id);
        printf("Grades: ");
        for (int j = 0; j < MAX_SUBJECTS; j++) {
            printf("%.2f ", students[i].grades[j]);
        }
        printf("\nAverage: %.2f\n\n", students[i].average);
    }

    return 0;
}

File: salida/archivo4.rs
const MAX_STUDENTS: usize = 3;
const MAX_SUBJECTS: usize = 3;
struct Student {
	name: [char ;50],
	
	let id:i32 ,
	
	grades: [f32 ;MAX_SUBJECTS],
	
	let average:f32 ,
	
}
fn calculateAverage (student: &mut Student) {
	let sum:f32 =0;
	
	for i in 0.. MAX_SUBJECTS{
		sum += student1.grades[i];
		
		
	}student.average:=sum / MAX_SUBJECTS;
	
	
}

fn () main-> i32 {
	students: [Student;MAX_STUDENTS];
	
	for i in 0.. MAX_STUDENTS{
		print!("Enter name of student %d: {}",i + 1);let mut input = String::new();
		io::stdin()
		.read_line(&mut input)
		.expect("Failed to read line");print!("Enter id of student %d: {}",i);let mut input = String::new();
		io::stdin()
		.read_line(&mut input)
		.expect("Failed to read line");print!("Enter grades for student %d:\n{}",i + 1);for j in 0.. MAX_SUBJECTS{
			print!("Enter grade for subject %d: {}",j + 1);let mut input = String::new();
			io::stdin()
			.read_line(&mut input)
			.expect("Failed to read line");
		}calculateAverage(&students[i]);
		
		
	}print!("\n\nDisplaying Information:\n\n");for i in 0.. MAX_STUDENTS{
		print!("Name: %s\n{}",students[i].name);print!("ID: %d\n{}",students[i].id);print!("Grades: ");for j in 0.. MAX_SUBJECTS{
			print!("%.2f {}",students[i].grades[j]);
		}print!("\nAverage: %.2f\n\n{}",students[i].average);
	}0
}

